sound/soc/sof/sof-audio.h

```diff
@@ -227,6 +228,14 @@ const struct sof_ipc_pipe_new *snd_sof_pipeline_find(struct snd_sof_dev *sdev,
 void snd_sof_pcm_period_elapsed(struct snd_pcm_substream *substream);
 void snd_sof_pcm_period_elapsed_work(struct work_struct *work);
 
+#if defined(CONFIG_SND_SOC_SOF_COMPRESS)
+void snd_sof_compr_fragment_elapsed(struct snd_compr_stream *cstream);
+void snd_sof_compr_fragment_elapsed_work(struct work_struct *work);
+#else
+static inline void snd_sof_compr_fragment_elapsed(struct snd_compr_stream *cstream) { }
+static inline void snd_sof_compr_fragment_elapsed_work(struct work_struct *work) { }
```

Comment:
making a "work" function `inline` makes no sense - you need its pointer. So, you can remove `inline` which in itself is a bit dubious: normally you don't implement (static) non-inline functions in headers. But TBH I'm not finding a better solution for this either, you want to avoid preprocessor conditionals in .c code as much as possible...

Reply:
This is the price we pay for having a clean '*.c' file. Removing inline I think will make the code fail to compile. Also, this indeed is very interesting because what I need is just a function pointer. 
Will give it some thoughts to this, maybe I can find a similar situation in the Linux kernel.

Reply:
@lyakh even if it is inline the function still has an address, so I think the approach even a little bit 'forced' is correct.

Reply:
@dbaluta don't think so, it isn't the case that an inline function has an address. In cases like this the compiler recognises that it needs an address of that function so effectively it ignores the `inline` qualifier and makes a proper function of it. At least for that specific use. If you had more uses of that function which didn't require its address and called it directly instead, maybe in those cases the compiler would accept your request and inline that function. But since you only have one use, effectively the `inline` qualifier is completely redundant and ignored there.

Reply:
@lyakh what do you think about @ujfalusi suggestion?

Reply:
@dbaluta yes, @ujfalusi 's suggestion seems like a reasonable way to fix this to me.

Reply:
otoh, with the "static inline" the functions will be only non inline in the source file where they are referenced, in all other files they will be just dropped.
It would cause issues if two object file linked into the same module would use the same inline function from the header, but probably in that case due to the static it would be still fine.

But, it is true that it depends on the compiler to do the right thing.

Reply:
@ujfalusi I've implemented your suggestion. Will push the code today after testing the change.

Reply:
Done!